### PR TITLE
Added lifetime annotation according to stack overflow.

### PR DESCRIPTION
**Pull request by paulheinr**

### CODE REVIEW

File: src/experiments/metis_test.rs [hunk 1]
```diff
@@ -74,7 +74,7 @@ mod tests {
     use metis::option::Seed;
     use metis::{Graph, Idx};
 
-    use crate::experiments::metis_test::{partition, Link, Node};
```
Comment (paulheinr) [inline src/experiments/metis_test.rs:77]: Ahh, my code formatter in CLion did that - it's the Save Actions plugin. Are you using another one? I can switch to yours.

Comment (Janekdererste) [reply src/experiments/metis_test.rs:77]: I use rustfmt. It comes with the cargo toolchain. You can enable it in clion in settings -> Langauages & Frameworks -> Rust -> Rustfmt

File: src/routing/router.rs [hunk 2]
```diff
@@ -10,16 +11,20 @@ pub struct Router<'router> {
...
+    pub(crate) fn new(cch: &'router CCH, converter: &NetworkConverter) -> Router<'router> {
         Router {
-            server: Server::new(customize(cch, graph))
+            server: Server::new(customize(cch, &Router::create_owned_graph(&converter)))
         }
     }
 
     pub(crate) fn customize(&mut self, cch: &'router CCH, graph: &OwnedGraph) {
         self.server = Server::new(customize(cch, graph));
     }
```
Comment (Janekdererste) [inline src/routing/router.rs:23]: Very nice, that you figured out the lifetimes. The signature looks terribly complicated though. Would it be possible to use a type alias for this?

**Review by Janekdererste — APPROVED** [review]
Great.

File: src/routing/router.rs (hunk 2 above)
Comment (paulheinr) [reply src/routing/router.rs:23]: I'm not quite sure. We need the lifetime `'q`, so I tried this one here: 
```
type P<'q> = PathServerWrapper<'q, CustomizedBasic<'router, CCH>>;
pub(crate) fn query<'q>(
        &'q mut self,
        from: usize,
        to: usize,
    ) -> QueryResult<Self::P<'_>, Weight>
```
But the compiler complains `type must outlive the lifetime 'q`. Unfortunately, the explanation in the rust reference is not very detailed. So I don't really know what to try else.

Comment (Janekdererste) [reply src/routing/router.rs:23]: That's exactly what I meant. That makes sense, that we have to have the different lifetimes of the `PathServerWrapper` and the `CCH` and therefore have to keep it all in this signature. Thanks for trying!

**Pull request merged**